CMakeLists.txt

```diff
@@ -27,6 +27,76 @@ ELSE()
     SET(VER_BUILD 0)
 ENDIF()
 
+# Configuration of compilators
+## Compiler flags to add
+### Flags for GCC and Clang
+set(GNU_FLAGS_WARNING_YES "-Wall -Wextra -Wformat=2 -Wpedantic -Wshadow -Wmissing-declarations -Wfloat-equal")
+set(GNU_FLAGS_WARNING_NO  "-Wno-unused-command-line-argument -Wno-documentation -Wno-format-nonliteral")
+set(GNU_FLAGS_WARNING     "${GNU_FLAGS_WARNING_YES} ${GNU_FLAGS_WARNING_NO}")
+set(GNU_FLAGS_ERROR       "-pedantic-errors -Werror")
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU" OR
+    CMAKE_C_COMPILER_ID STREQUAL "GCC")
+  set(GNU_FLAGS_SUGGEST   "-Wsuggest-attribute=pure")
+endif()
+set(GNU_FLAGS_DEFAULT     "${GNU_FLAGS_WARNING} ${GNU_FLAGS_ERROR} ${GNU_FLAGS_SUGGEST}")
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU"   AND
+    CMAKE_C_COMPILER_VERSION VERSION_GREATER 4.9)
+  set(GNU_FLAGS_DEFAULT   "${GNU_FLAGS_DEFAULT} -fdiagnostics-color=auto")
+endif()
+set(GNU_FLAGS_OPTIMIZE    "-O2 -s")
+set(GNU_FLAGS_SECURITY_PI "-fPIC -fPIE -pie")
+if (( CMAKE_C_COMPILER_ID STREQUAL "GNU"   AND
+      CMAKE_C_COMPILER_VERSION VERSION_LESS 4.9 ) OR
+    ( CMAKE_C_COMPILER_ID STREQUAL "Clang" AND
+      CMAKE_C_COMPILER_VERSION VERSION_LESS 3.5 ))
+  set(GNU_FLAGS_SECURITY_STACK "-fstack-protector-all")
+else()
+  set(GNU_FLAGS_SECURITY_STACK "-fstack-protector-strong")
+endif()
+set(GNU_FLAGS_SECURITY_COMPILE "-D_FORTIFY_SOURCE=2")
+set(GNU_FLAGS_SECURITY_COMPILE
+  "${GNU_FLAGS_SECURITY_COMPILE} ${GNU_FLAGS_SECURITY_PI}")
+set(GNU_FLAGS_SECURITY_COMPILE
+  "${GNU_FLAGS_SECURITY_COMPILE} ${GNU_FLAGS_SECURITY_STACK}")
+set(GNU_FLAGS_SECURITY_LINK    "-Wl,-z,relro -Wl,-z,now")
```

Comment:
All these flags look very paranoid to me. Could you explain what is bad with the current defaults?
As for me, I'd rather make cmake use better default by provided patches for them if it's not already the case, and make this project use a more recent cmake version. We are kinda reinventing the wheel here to me.